/*function saludar(nombre, callback) {
  console.log(`Hola, ${nombre}`);
  callback();
}

function despedida() {
  console.log("Adi√≥s, nos vemos luego.");
}

saludar("Mar√≠a", despedida);

//
function mostrarMensaje(mensaje) {
  console.log(mensaje);
}

function procesar(callback) {
  console.log("Procesando datos...");
  callback("Datos procesados correctamente.");
}

procesar(mostrarMensaje);

const pedirComida = new Promise((resolve, reject) => {
  let ingredientes = Math.random() > 0.5;

  setTimeout(() => {
    if (ingredientes) {
      resolve("üçî Pedido listo: Hamburguesa.");
    } else {
      reject("‚ùå No hay ingredientes disponibles.");
    }
  }, 3000);
});

pedirComida
  .then((mensaje) => {
    console.log(mensaje);
  })

  .catch((error) => {
    console.error(error); // Si hubo un error
  });
  
*/
function hacerPedidoPizza() {
  return new Promise((resolve, reject) => {
    let ingredientes = Math.random() > 0.5;

    setTimeout(() => {
      if (ingredientes) {
        resolve("üçï Pedido listo: Pizza.");
      } else {
        reject("‚ùå No hay ingredientes para la pizza.");
      }
    }, 4000);
  });
}

async function hacerPedido() {
  try {
    console.log("‚è≥ Esperando el pedido...");
    let mensaje = await hacerPedidoPizza();
    console.log(mensaje);
  } catch (error) {
    console.error(error);
  }
}

hacerPedido();

